app/scripts/lib/AbstractPetnamesBridge.ts

```diff
@@ -0,0 +1,240 @@
+import {
+  NameController,
+  NameStateChange,
+  NameType,
+  SetNameRequest,
+} from '@metamask/name-controller';
+import { RestrictedControllerMessenger } from '@metamask/base-controller';
+
+// Use the same type for both the source entries and the argument to NameController::setName.
+export type PetnameEntry = SetNameRequest;
+
+// The type of change that occurred.
+export enum ChangeType {
+  ADDED = 'ADDED',
+  UPDATED = 'UPDATED',
+  DELETED = 'DELETED',
+}
+
+type SyncDirection = 'Source->Petnames' | 'Petnames->Source';
```

Comment:
As with the `ChangeType`, is it more idiomatic to use enums for fixed strings rather unions?

Reply:
Done in next update.